src/helpers/screenings.js

```diff
@@ -3,13 +3,52 @@ import fetch from "node-fetch";
 const API = "https://plankton-app-xhkom.ondigitalocean.app/api/screenings";
 
 export async function movieScreenings(id) {
-    const res = await fetch(`${API}?filters[movie]=${id}`);
-    const payload = await res.json();
-    return payload.data;
+  const res = await fetch(`${API}?filters[movie]=${id}`);
+  const payload = await res.json();
+  return payload.data;
 }
 
 export async function getAllScreenings() {
-    const res = await fetch(API);
-    const payload = await res.json();
-    return payload.data;
+  const res = await fetch(API);
+  const payload = await res.json();
+  return payload.data;
+}
+
+export async function screeningsStartpage() {
+  const apiRes = await fetch(API + "?populate=movie");
+  const payload = await apiRes.json();
+  const result = payload.data
+    .map((item) => ({
+      id: item.id,
+      ...item.attributes,
+      movie: {
+        id: item.attributes.movie.data.id,
+        ...item.attributes.movie.data.attributes,
+      },
+    }))
+    .filter((screenings) => {
+      const getDatesBetweenDates = (startDate, endDate) => {
+        let dates = [];
+
+        const theDate = new Date(startDate);
+        while (theDate < new Date(endDate)) {
+          dates = [...dates, new Date(theDate).toISOString().split("T")[0]];
+          theDate.setDate(theDate.getDate() + 1);
+        }
+        dates = [...dates, new Date(endDate).toISOString().split("T")[0]];
+        return dates;
+      };
+      const start = new Date().toISOString().split("T")[0];
+      const end = new Date(new Date().getTime() + 4 * 24 * 60 * 60 * 1000)
+        .toISOString()
+        .split("T")[0];
+
+      for (let i = 0; i < getDatesBetweenDates(start, end).length; i++) {
+        if (screenings.start_time.includes(getDatesBetweenDates(start, end)[i])) {
```

Comment:
There are easier ways of doing this. I would use a completely different logic, using `Date.getTime()` and simple comparison with the `<` and `>` operators.

But if you do want to use this approach of creating an array of dates, and seeing if a certain date is among those dates, an easier way would be to just do `getDatesBetweenDates().includes(screenings.start_time)`, without the for loop.

You would of course then have to make sure that the format is identical for all of the data in the array, and `screenings.start_time`.